zephyr/esp32s3/src/wifi/esp_wifi_adapter.c

```diff
@@ -36,6 +36,8 @@ LOG_MODULE_REGISTER(esp32_wifi_adapter, CONFIG_WIFI_LOG_LEVEL);
 #include "soc/dport_reg.h"
 #include "driver/adc2_wifi_private.h"
 
+#define WIFI_PKT_SIZE	1700
+K_HEAP_DEFINE(wifi_heap, WIFI_PKT_SIZE*MAX(CONFIG_NET_BUF_RX_COUNT, CONFIG_NET_BUF_TX_COUNT));
```

Comment:
I added the WiFi packet size of 1700 based upon the allocation requests.  It seems that for a 1500 byte Ethernet packet, the allocation by the ESP WiFi library is 1700 bytes.

Ideally, we should allocate enough buffers for both RX and TX packets, but that is a large amount of memory and does not appear to improve performance in my testing.

Reply:
This heap is also used to allocate the message queue and a few other OS items, so not sure if there should be a base allocation amount to cover that overhead or if those should be allocated from the system heap.  It may be that this buffer needs to be a separate Kconfig item with a range of buffers.

It could also be the case of thread priority issues that are causing the excessive buffer usage and maybe the best solution is to have a small buffer here and to fix the priority issues.  That would require some profiling and tracing which I don't have time for.  Let me change this PR to a draft while it gets discussed.